celery/app/base.py

```diff
@@ -1149,24 +1151,25 @@ def current_worker_task(self):
         """
         return get_current_worker_task()
 
-    @cached_property
+    @property
     def oid(self):
         """Universally unique identifier for this app."""
-        # since 4.0: thread.get_ident() is not included when
-        # generating the process id.  This is due to how the RPC
-        # backend now dedicates a single thread to receive results,
-        # which would not work if each thread has a separate id.
-        return oid_from(self, threads=False)
+
+        if not hasattr(tlocal, 'oid'):
+            tlocal.oid = oid_from(self, threads=True)
+        return tlocal.oid
```

Comment:
May as well `try, except AttributeError` here instead of doing a `hasattr`.
```suggestion
        try:
            return tlocal.oid
        except AttributeError:
            tlocal.oid = new_oid = oid_from(self, threads=True)
            return new_oid
```
Same applies below.